src/Avalonia.X11/Interop/Glib.cs

```diff
@@ -0,0 +1,192 @@
+#nullable enable
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Avalonia.Platform.Interop;
+using static Avalonia.X11.Interop.Glib;
+namespace Avalonia.X11.Interop;
+
+internal static unsafe class Glib
+{
+    private const string GlibName = "libglib-2.0.so.0";
+    private const string GObjectName = "libgobject-2.0.so.0";
+
+    [DllImport(GlibName)]
+    public static extern void g_slist_free(GSList* data);
+
+    [DllImport(GObjectName)]
+    private static extern void g_object_ref(IntPtr instance);
+
+    [DllImport(GObjectName)]
+    private static extern ulong g_signal_connect_object(IntPtr instance, Utf8Buffer signal,
+        IntPtr handler, IntPtr userData, int flags);
+
+    [DllImport(GObjectName)]
+    private static extern void g_object_unref(IntPtr instance);
+
+    [DllImport(GObjectName)]
+    private static extern ulong g_signal_handler_disconnect(IntPtr instance, ulong connectionId);
+
+    public const int G_PRIORITY_HIGH = -100; 
+    public const int G_PRIORITY_DEFAULT = 0; 
+    public const int G_PRIORITY_HIGH_IDLE =  100; 
+    public const int G_PRIORITY_DEFAULT_IDLE =  200; 
+    
+    [DllImport(GlibName)]
+    public static extern IntPtr g_main_loop_new(IntPtr context, int is_running);
+    
+    [DllImport(GlibName)]
+    public static extern void g_main_loop_quit(IntPtr loop);
+    
+    [DllImport(GlibName)]
+    public static extern void g_main_loop_run(IntPtr loop);
+    
+    [DllImport(GlibName)]
+    public static extern void g_main_loop_unref(IntPtr loop);
+
+    [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+    public delegate void GOnceSourceFunc(IntPtr userData);
```

Comment:
I noticed you avoided runtime marshalling in these definitions. But just in case, delegates with `UnmanagedFunctionPointer` are still not supported there, you can only use `UnmanagedCallersOnly` statics.

![image](https://github.com/user-attachments/assets/6999a3b5-d7d6-444b-9ef8-f5f6138533ad)

Reply:
But since the rest of X11 backend is not compatible anyway, I don't think it's important problem for this PR.

Reply:
Not sure what do you mean by "avoided runtime marshalling". I've just placed delegates to static fields to make sure that they won't get collected. It might be not necessary with the recent Roslyn though